### PR TITLE
legal, SIWE, and fix session bug

### PR DESCRIPTION
**Pull request by tehnalogos**

### CODE REVIEW

File: frontend/app/terms/page.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+          The revival of assets from the GRAVE back to your Universal Profile is
+          subject to a fee. The specific fee structure is as follows:
+        </Text>
+        <List spacing={2} mt={4}>
+          <ListItem>
+            A fixed fee of 0.005 LYX is charged per revival action.
+          </ListItem>
+          <ListItem>
+            Fees are subject to change at our discretion. We will endeavor to
+            notify you of any fee changes in a timely manner.
```
Comment (Ledespana) [inline frontend/app/terms/page.tsx:113]: Do we want to actually notify anyone? should we remove that?

File: frontend/app/terms/page.tsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        {/* Intellectual Property Rights */}
+        <Heading id="ip" as="h1" size="lg" mb={4} mt={20}>
+          Intellectual Property Rights
+        </Heading>
+        <Heading as="h2" size="md" mt={10}>
+          1. Ownership
+        </Heading>
+        <Text mt={4}>
+          Universal Grave, its original content, features, and functionality are
+          and will remain the exclusive property of [Your Company Name] and its
```
Comment (Ledespana) [inline frontend/app/terms/page.tsx:170]: [Your Company Name]

I feel that is on purpose for the future, but just checking

File: frontend/app/terms/page.tsx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+          licensors. Our trademarks and trade dress may not be used in
+          connection with any product or service without the prior written
+          consent of [Your Company Name].
```
Comment (Ledespana) [inline frontend/app/terms/page.tsx:173]: [Your Company Name]

File: frontend/app/terms/page.tsx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+        {/* Limitation of Liability */}
+        <Heading id="limitations" as="h1" size="lg" mb={4} mt={20}>
+          Limitation of Liability
+        </Heading>
+        <Heading as="h2" size="md" mt={10}>
+          1. Limitation Clause
+        </Heading>
+        <Text mt={4}>
+          In no event shall [Your Company Name], nor its directors, employees,
```
Comment (Ledespana) [inline frontend/app/terms/page.tsx:184]: [Your Company Name]

File: frontend/app/terms/page.tsx [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+        >
+          Modification and Termination
+        </Heading>
+        <Heading as="h2" size="md" mt={10}>
+          1. Right to Modify or Discontinue
+        </Heading>
+        <Text mt={4}>
+          We reserve the right to modify or discontinue, temporarily or
+          permanently, the Service (or any part thereof) with or without notice
+          at any time. You agree that [Your Company Name] shall not be liable to
```
Comment (Ledespana) [inline frontend/app/terms/page.tsx:215]: [Your Company Name]

File: frontend/components/wallet/WalletProvider.tsx [hunk 6]
```diff
@@ -100,27 +105,37 @@ export const WalletProvider: React.FC<Props> = ({ children }) => {
...
+            `${window.location.host}/terms#privacy`,
+            `${window.location.host}/terms#fees`,
+          ], // Information the user wishes to have resolved as part of authentication by the relying party
         }).prepareMessage();
-        const hashedMessage = web3.eth.accounts.hashMessage(siweMessage);
-
+        const signature = await web3.eth.personal.sign(
+          siweMessage,
+          accounts[0],
+          'test password!'
```
Comment (Ledespana) [inline frontend/components/wallet/WalletProvider.tsx:126]: test password?

File: frontend/app/terms/page.tsx (hunk 2 above)
Comment (tehnalogos) [reply frontend/app/terms/page.tsx:170]: yes ser. there's another ticket handling this for the future

File: frontend/components/wallet/WalletProvider.tsx (hunk 6 above)
Comment (hosainnet) [reply frontend/components/wallet/WalletProvider.tsx:126]: do we just have to put something here?

**Review by Ledespana — APPROVED** [review]

**Review by democrazyfi — APPROVED** [review]

**Pull request merged**